src/EditorFeatures/Core/Wrapping/WrapItemsAction.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using static Microsoft.CodeAnalysis.CodeActions.CodeAction;
+
+namespace Microsoft.CodeAnalysis.Editor.Wrapping
+{
+    /// <summary>
+    /// Code action for actually wrapping items.  Provided as a special subclass because it will
+    /// also update the wrapping most-recently-used list when the code action is actually
+    /// invoked.
+    /// </summary>
+    internal class WrapItemsAction : DocumentChangeAction
+    {
+        // Keeps track of the invoked code actions.  That way we can prioritize those code actions 
+        // in the future since they're more likely the ones the user wants.  This is important as 
+        // we have 9 different code actions offered (3 major groups, with 3 actions per group).  
+        // It's likely the user will just pick from a few of these. So we'd like the ones they
+        // choose to be prioritized accordingly.
+        private static ImmutableArray<string> s_mruTitles = ImmutableArray<string>.Empty;
+
+        public string ParentTitle { get; }
+
+        public string SortTitle { get; }
+
+        public WrapItemsAction(string title, string parentTitle, Func<CancellationToken, Task<Document>> createChangedDocument)
+            : base(title, createChangedDocument)
+        {
+            ParentTitle = parentTitle;
+            SortTitle = parentTitle + "_" + title;
+        }
+
+        protected override Task<IEnumerable<CodeActionOperation>> ComputePreviewOperationsAsync(CancellationToken cancellationToken)
+        {
+            // For preview, we don't want to compute the normal operations.  Specifically, we don't
+            // want to compute the stateful operation that tracks which code action was triggered.
+            return base.ComputeOperationsAsync(cancellationToken);
+        }
+
+        protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(CancellationToken cancellationToken)
+        {
+            var operations = await base.ComputeOperationsAsync(cancellationToken).ConfigureAwait(false);
+            var operationsList = operations.ToList();
+
+            operationsList.Add(new RecordCodeActionOperation(this.SortTitle, this.ParentTitle));
+            return operationsList;
+        }
+
+        public static ImmutableArray<CodeAction> SortActionsByMostRecentlyUsed(ImmutableArray<CodeAction> codeActions)
+        {
+            // make a local so this array can't change out from under us.
+            var mruTitles = s_mruTitles;
+            return codeActions.Sort((ca1, ca2) =>
```

Comment:
are there any other code action having the list ordering decide themselves?

Reply:
Yes.   For example, when you generate a member, we'll use the capitalization of the name to determine if we shoudl offer things like "generate field" above "generate property" (or vice versa).  We order things to provide the most likely and most valuable item to the user nearer to the top.